src/VisualStudio/Core/Def/Options/PackageSettingsPersister.cs

```diff
@@ -42,12 +43,19 @@ private async Task InitializeAsync(CancellationToken cancellationToken)
 
             _lazyRoslynPackage = await RoslynPackage.GetOrLoadAsync(_threadingContext, _serviceProvider, cancellationToken).ConfigureAwait(true);
             Assumes.Present(_lazyRoslynPackage);
-
-            _optionService.RefreshOption(new OptionKey2(SolutionCrawlerOptionsStorage.SolutionBackgroundAnalysisScopeOption), _lazyRoslynPackage.AnalysisScope);
             _lazyRoslynPackage.AnalysisScopeChanged += OnAnalysisScopeChanged;
+
+            await TaskScheduler.Default;
```

Comment:
📝 This would introduce a race condition, since `AnalysisScopeChanged` is set from the UI thread it could potentially be set between the event hookup and the explicit call.